src/tools/csg_resample.cc

```diff
@@ -171,7 +171,7 @@ int main(int argc, char **argv) {
           if (in.x(i) < sp_min) {
             minindex = i;
           }
-          if (in.x(i) < sp_max) {
+          if (in.x(i) <= sp_max) {
```

Comment:
`=` comparison for float is usually not a good idea. @JensWehner what would you do here?

Reply:
Yes I am not really sure. In real examples it should not matter so much and we rely a bit on a corner case here. I am not sure atm

Reply:
I think the major question is what should a range 1:10 mean is it incluive or exclusive.

Reply:
@JensWehner  How about `in.x(i) < sp_max * (1 + DBL_EPSILON)`? (see http://www.cplusplus.com/reference/cfloat/).

Reply:
> I think the major question is what should a range 1:10 mean is it incluive or exclusive.

Inclusive.

Reply:
if we do it like this I would use https://en.cppreference.com/w/cpp/types/numeric_limits instead of the c way. i.e. `std::numerics<Index>::epsilon()`. Why not `sp_max+std::numerics<Index>::epsilon()`, although I think that would be equivalent to `in.x(i) <= sp_max`